### ISSUE TITLE
Android media service not being started by just_audio_background

### ISSUE THREAD

**Issue by efraimbart** [ryanheise/just_audio#543]
**Which API doesn't behave as documented, and how does it misbehave?**
The just_audio_background Android media service doesn't appear to start when media is played, as evidenced by the notification not showing up. 

**Minimal reproduction project**
The Example

**To Reproduce (i.e. user steps, not code)**
1. Checkout the example repo
2. Run `flutter run`
3. Click play in-app
4. Notice no music notification

**Error messages**

**Expected behavior**
Android notification service should always start when media is played

**Screenshots**
![Screenshot_1633722799](https://user-images.githubusercontent.com/8098171/136618062-f8dae145-88a8-4352-a8d8-15eddc775924.png)
![Screenshot_1633722803](https://user-images.githubusercontent.com/8098171/136618066-ef3c603b-89df-400d-ad56-b7cd9e83eaef.png)

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**
 - Device: Pixel 5 emulator
 - Device: Samsung Galaxy Tab S7+
 - Device: Samsung Galaxy Note20 Ultra 5G
 - OS: Android 11 (30)

**Flutter SDK version**
```
[√] Flutter (Channel stable, 2.5.0, on Microsoft Windows [Version 10.0.22000.194], locale en-US)
[√] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
[√] Chrome - develop for the web
[√] Visual Studio - develop for Windows (Visual Studio Community 2019 16.11.2)
[√] Android Studio (version 2020.3)
[√] VS Code (version 1.60.0)
[√] Connected device (5 available)
```

**Additional context**
This doesn't seem to always happen. On the release version of my app using just_audio_background it works fine, but when I try to run the example app it does not work. And sometimes it stops working for my app as well, when debugging. Perhaps there is some sort of conflict when there are multiple packages using just_audio_background at the same time?

This doesn't appear to be an issue with the audio_service example.

The reason I am testing this is because I may have found a bug with stopping music playback from the notification on Android, but when I started attempting to test with the example app things went haywire.

**Comment by efraimbart** [comment]
It's happening for me 100% of the time when using flutter run on both the example app (`just_audio_background/example/lib/main.dart`) and my personal app. It doesn't happen in my personal app when I install the Google Play version built with GitHub actions.

Perhaps you can try running another instance of the example app with a different package name simultaneously and see if that has any effect?

I'll try and get you some logs, and perhaps even test on a new environment to see how to best replicate. I'll try to update to the new audio service version, but as you said, I doubt it will change anything as the audio service example works fine.

**Comment by efraimbart** [comment]
Apparently the example is already pulling audio_service 0.18.0

**Comment by ryanheise** [comment]
That would happen if you've manually pulled 0.18.0.

However, regarding the error above, it indicates that the Android resource you're using for one of the icons isn't in a valid format, containing a slash. For example, `drawable/audio_service_pause`.

I don't understand how this error could happen in the example, because all resource names supplied are valid???

**Comment by efraimbart** [comment]
My bad, apparently when I was digging through the source code of just_audio_background I changed the value for one of the audio_service's `MediaControl` items. I guess .pub_cache is shared between all flutter projects - and possibly doesn't even get cleared with `flutter clean`? - causing me to see the bug I caused across all projects using just_audio. 

Sorry about that, and thanks for pointing me in the right direction!

I'll create a new issue for the bug that I was originally looking into, which is apparently present in the example as well.
